src/ax/saga/persistence/eventsourcing/snapshot.go

```diff
@@ -32,4 +32,16 @@ type SnapshotRepository interface {
 		pk string,
 		i saga.Instance,
 	) error
+
+	// DeleteSagaSnapshots deletes any snapshots associated with a saga instance.
+	//
+	// The implementation may return an error if snapshots for this instance
+	// already exists, but belong to a different saga, as identified by pk, the
```

Comment:
An error if the snapshots for this instance already exist? Is that correct?

Reply:
It is trying to delete them.

Reply:
It's correct, but maybe taking a way the word "already" would be good:

```go
	// It returns an error if the instance exists but belongs to a
	// different saga, as identified by pk, the saga's persistence key.
```

Reply:
Or even just "if the instance belongs to a different saga"